src/librekey/key_store_g10.c

```diff
@@ -808,17 +1427,53 @@ rnp_key_store_g10_key_to_mem(pgp_io_t *     io,
 
     case PGP_PTAG_CT_SECRET_KEY:
     case PGP_PTAG_CT_SECRET_SUBKEY:
-        if (!add_block_to_sexp(&s_exp, (uint8_t *) "private-key", sizeof("private-key") - 1)) {
-            return false;
+        if (key->key.seckey.alg == PGP_SA_PLAINTEXT) {
+            if (!add_string_block_to_sexp(&s_exp, "private-key")) {
+                return false;
+            }
+        } else {
+            if (!add_string_block_to_sexp(&s_exp, "protected-private-key")) {
+                return false;
+            }
+
+            // force switch to AES with CBC and SHA-1
+            if (key->key.seckey.alg != PGP_SA_AES_128) {
+                key->key.seckey.alg = PGP_SA_AES_128;
+            }
+
+            if (key->key.seckey.block_cipher_mode != PGP_BLOCK_CIPHER_MODE_CBC) {
+                key->key.seckey.block_cipher_mode = PGP_BLOCK_CIPHER_MODE_CBC;
+            }
+
+            if (key->key.seckey.hash_alg != PGP_HASH_SHA1) {
```

Comment:
Yes, it is a hack. And this hack has a reason: to save any algorithm/cipher, I should implement `opengpg-native` mode of encoding that I haven't done yet.

So, I see few diffferent ways:
 - remove it and disable G10 for default key settings;
 - update default settings;
 - keep it as is.

Anyway, I will remove it as next step with `openpgp-native`.